host_core/lib/host_core/host.ex

```diff
@@ -204,7 +212,8 @@ defmodule HostCore.Host do
       lattice_rpc_url: "",
       provider_key: provider_key,
       # TODO
-      env_values: %{}
+      env_values: %{},
+      invocation_seed: cluster_seed()
```

Comment:
Having a little bit of trouble tracking this and feeling good about it, so here for the hostinfo we are passing the cluster seed key over to the providers. Is the cluster seed a private key that we are okay handing out to the providers? If (when?) we implement provider discovery, wouldn't any provider that has access to the lattice also have access to the cluster key anyways?

Let me know if this cluster seed is the public keypair, or if I'm thinking about what the cluster should be incorrectly.

Reply:
In this situation we are passing the cluster seed to the provider because this provider is _trusted_ by virtue of us a) having access to the PAR file and b) being the ones who started it in the first place. So, we give it a seed (private) key so that it can sign invocations bound for any host that has the public half of that in the list of valid issuers.

We don't trust external (not even implemented yet) providers, so we will _not_ give them a signing key - they'll have to be given one by whatever process started them in the first place. Access to the signing key _without the host runtime providing it_ is essentially what guarantees the authenticity of the remote/external providers.

Reply:
I don't feel like this is the best solution, but it's **a** solution right now. If we are going to protect invocations by requiring them to be signed by one of the valid/known issuers, then we have to give the providers we auto-launch **a** signing key, whereas external providers we assume they've gained access to a key because they deserve it.

In an ideal world there would be a simpler way to prevent unauthorized access to the lattice once inside a NATS connection (defense in depth), but I don't yet know of a simpler way than signing invocations.

Reply:
this makes a good argument for (eventually when supported) using an OPA policy that allow-lists certain provider issuers because if you allow _any_ issuer, then anyone could put up a bogus provider in a public registry and if they gained access to the control interface, they could request the starting of the trojan PARjeezy, which could then exfiltrate the current signing key.

This is still pretty far-remote in terms of likelihood because for this attack vector to work, the attacker needs to compromise **three** things - NATS access to the control interface, NATS access to the RPC interface, and publish a trojan capability provider to a registry that the environment has access to.

Reply:
Okay, I'm fine with it for now then, I know we discussed the cluster signing as a mechanism for defense in depth so I just wanted to understand the implementation 👍🏻 

We'll just want people to keep in mind that if they are starting a provider in their lattice, that you have to trust it (OPA will be good for this) and it's not the same level of security that WASM guarantees re: not accessing information it's not supposed to